project/XDEv11/README.md

```diff
@@ -0,0 +1,70 @@
+# Maximum Graph Matching Library (MGML)
+
+## Basic Information
+
+In graph theory, a matching is a set of non-adjacent edges in a graph,
+so the maximum matching problem is to find the maximum set in terms of
+cardinality or weight.
+
+## Problem to Solve
+
+There are different algorithms for different situations. This project
+is trying to implement some popular algorithms in C++ and to also
+provide a API in python. Here are the problems I am going to solve.
+
+* Maximum Cardinality Bipartite Matching
+	* Hopcroft-Karp Algorithm ($O(|E|\sqrt{|V|})$)
+
+* Maximum Weight Bipartite Matching
+	* Hungarian Algorithm ($O(|V|^3)$)
+
+* Maximum Cardinality Matching in arbitrary graphs
+	* Blossom Algorithm ($|V||E|\alpha(|V|)$)
+	* MV Matching Algorithm (**optional**)
+
+* Maximum Weight Matching in arbitrary graphs
+	* Blossom Algorithm (**optional**)
+
+## Prospective Users
+
+Anyone who needs the answer of maximum matching problem.
+
+## System Architecture
+
+The algorithms are implemented in C++ for fast runtime, and there are
+interfaces in C++ for sure and in python by pybind11.
+
+## API Description
+
+The input will consist of a graph represented in adjacency list or
+adjacency matrix, and the output will be a matching.
+
+## Engineering Infrastructure
+
+* Build: GNU Make
+
+* Version control: Git (GitHub)
+
+* Testing: unittest (python)
+
+* Document: Markdown (README.md)
+
+## Schedule
+
+Week 1 (4/17): Design interface and architecture
+
+Week 2 (4/24): Write some unit tests
+
+Week 3 (5/01): Implement Hopcroft-Karp Algorithm
```

Comment:
Interface and architecture usually emerges after you have a data structure for the algorithms.

3 algorithms in a project are not few.  Please carefully evaluate the schedule for the final presentation.

Reply:
The data structure is adjacency list or adjacency matrix for a graph. However, I think I should provide some data types.

I'll take care of the time.